libursa/DatabaseConfig.cpp

```diff
@@ -0,0 +1,44 @@
+#include "DatabaseConfig.h"
+
+#include "spdlog/spdlog.h"
+
+DatabaseConfig::DatabaseConfig(json config) : config_(config) {
+    defvals_ = {
+        {"query_max_edge", 1},
+        {"query_max_ngram", 16},
+    };
+    for (const auto &elm : config.items()) {
+        if (defvals_.count(elm.key()) == 0) {
+            spdlog::warn("Unexpected config: {}={}", elm.key(),
+                         elm.value().dump(4));
+        } else {
+            spdlog::info("Config: {}={}", elm.key(), elm.value().dump(4));
+        }
+    }
+}
+
+uint64_t DatabaseConfig::get(const ConfigKey &key) const {
+    if (defvals_.count(key.key()) == 0) {
+        throw std::runtime_error("Invalid config key (get)");
+    }
+    const auto &it = config_.find(key.key());
+    if (it == config_.end()) {
+        return defvals_.at(key.key());
+    }
+    return *it;
+}
+
+void DatabaseConfig::set(const ConfigKey &key, uint64_t value) {
+    if (defvals_.count(key.key()) == 0) {
+        throw std::runtime_error("Invalid config key (set)");
+    }
+    config_[key.key()] = value;
+}
+
+std::map<std::string, uint64_t> DatabaseConfig::get_all() const {
+    std::map<std::string, uint64_t> result;
```

Comment:
```suggestion
    std::unordered_map<std::string, uint64_t> result;
```

This change requires `#include <unordered_map>`!